src/exercises/cheriabi/README.md

```diff
@@ -38,24 +38,25 @@ evasion of CHERI capabilities' protection properties, as CHERI capabilities are
 interpreted in combination with the virtual memory map.
 
 However, in CheriBSD, the kernel can return memory mappings (from `mmap`) using
-a bounded capability.  Moreover, it can demand that references to memory
-mappings passed to other system calls (such as `munmap`, `madvise`, or even
-`mmap` itself) be made using capabilities.  This reduces ambient authority
-significantly: software must have a capability to a part of the memory map
-rather than merely the relevant address.
+a bounded capability.  Moreover, it demands that references to memory mappings
+passed to other system calls (such as `munmap`, `madvise`, or even `mmap`
+itself) be made using capabilities.  This reduces ambient authority
+significantly: software must provide a capability to a part of the memory map
+rather than simply passing its integer address.
 
 And yet, that may not be sufficient.  It implies that any capability to a part
 of the address space (or, at least, whole pages thereof) implicitly carries the
 authority to arbitrarily manipulate that part of the address space.  This is
 likely undesirable; for example, we likely do not mean or expect *clients* of
-`malloc` to be able to, for example, *unmap* or *alias* parts of the heap!
+`malloc` to be able to, for example, *unmap* or *alias* parts of the heap!  That
+is, clients of `malloc` should not have sufficient privilege to manipulate the
+definition of the address space being used to back the heap.
```

Comment:
When implementing the VMEM permission in the first place, my internal argument was that being able to change the backing of a mapping potentially violated monotonicity unless you had explicit permission to do so since you can't know if an allocation has been subdivided.